### PR TITLE
feat: Demo mode

### PR DESCRIPTION
**Pull request by Matvey-Kuk**
LIVE_DEMO_MODE will launch what uset to be simulate_alert.py in the background

Closes https://github.com/keephq/keep/issues/2524

✅ Closed in favor of https://github.com/keephq/keep/pull/2561

## Summary by CodeRabbit

- **New Features**
	- Introduced a new environment variable `KEEP_LIVE_DEMO_MODE` for simulating alerts in demo mode.
	- Added functionality for simulating alerts through the enhanced `demo_mode_runner` module, including structured correlation rules and a robust service topology.

- **Bug Fixes**
	- Enhanced alert configurations across multiple providers to improve specificity and organization.

- **Documentation**
	- Updated deployment documentation to include the new `KEEP_LIVE_DEMO_MODE` environment variable.

- **Chores**
	- Simplified alert simulation logic by integrating with the new demo mode functionality.

### CODE REVIEW

File: keep/api/core/demo_mode_runner.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+correlation_rules_to_create = [
+    {
+        "sqlQuery": {
+            "sql": "((name like :name_1))",
+            "params": {
+                "name_1": "%mq%"
+            }
+        },
+        "groupDescription": "This rule groups all alerts related to MQ.",
+        "ruleName": "Message Queue Bucle Up",
```
Comment (talboren) [inline keep/api/core/demo_mode_runner.py:42]: Bucle?

File: keep/api/core/demo_mode_runner.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                logger.error("Failed to send alert: {}".format(response.text))
+            else:
+                logger.info("Alert sent successfully")
+
+def launch_demo_mode():
+    """
+    Running async demo in the backgound.
+    """
+    keep_api_url = "http://localhost:" + str(os.environ.get("PORT", 8080))
+    keep_api_key = get_or_create_api_key(
```
Comment (talboren) [inline keep/api/core/demo_mode_runner.py:166]: can be set from env variable

File: keep/api/core/demo_mode_runner.py [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+
+def get_or_create_topology(keep_api_key, keep_api_url):
+    services_existing = requests.get(
+        f"{keep_api_url}/topology",
+        headers={"x-api-key": keep_api_key},
+    )
+    services_existing.raise_for_status()
+    services_existing = services_existing.json()
+
+    if len(services_existing) == 0 or True:
```
Comment (talboren) [inline keep/api/core/demo_mode_runner.py:233]: because of "or True" it'll always run, I think

File: keep/api/config.py [hunk 4]
```diff
@@ -20,6 +21,7 @@ def on_starting(server=None):
 
     migrate_db()
     launch_uptime_reporting()
+    launch_demo_mode()
```
Comment (talboren) [inline keep/api/config.py:24]: wdyt if
```suggestion
    if LIVE_DEMO_MODE:
        launch_demo_mode()
```
?

Comment (Matvey-Kuk) [reply keep/api/config.py:24]: up to you, wanted to manage the env var within the module :)

Comment (talboren) [reply keep/api/config.py:24]: again, it's just me being extra cautious with these things we plug in that are not "production". I would even go further with:
```suggestion
    if LIVE_DEMO_MODE:
        from keep.api.core.demo_mode_runner import launch_demo_mode
        launch_demo_mode()
```
but your call on this one

File: keep/api/core/demo_mode_runner.py (hunk 3 above)
Comment (Matvey-Kuk) [reply keep/api/core/demo_mode_runner.py:233]: gooooooood catch!

File: keep/api/config.py [hunk 5]
```diff
@@ -13,6 +13,8 @@
 keep.api.logging.setup_logging()
 logger = logging.getLogger(__name__)
 
+LIVE_DEMO_MODE = os.environ.get("LIVE_DEMO_MODE", "false").lower() == "true"
```
[the remark this answers is not in the document]
Comment (talboren) [reply keep/api/config.py:16]: just fo the convention <3

File: docs/deployment/configuration.mdx [hunk 6]
```diff
@@ -25,6 +25,7 @@ General configuration variables control the core behavior of the Keep server. Th
 | **KEEP_API_URL** | Specifies the Keep API URL | No | Constructed from HOST and PORT | Valid URL |
 | **KEEP_STORE_RAW_ALERTS** | Enables storing of raw alerts | No | "false" | "true" or "false" |
 | **TENANT_CONFIGURATION_RELOAD_TIME** | Time in minutes to reload tenant configurations | No | 5 | Positive integer |
+| **LIVE_DEMO_MODE** | Keep will simulate incoming alerts and other activity | No | "false" | "true" or "false" |
```
Comment (talboren) [inline docs/deployment/configuration.mdx:28]:
```suggestion
| **KEEP_LIVE_DEMO_MODE** | Keep will simulate incoming alerts and other activity | No | "false" | "true" or "false" |
```

File: keep/api/core/demo_mode_runner.py [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+
+
+def launch_demo_mode():
+    """
+    Running async demo in the backgound.
+    """
+    logger.info("Demo mode launched.")
+
+    keep_api_url = "http://localhost:" + str(os.environ.get("PORT", 8080))
+    keep_api_key = os.environ.get("KEEP_API_KEY")
```
Comment (talboren) [inline keep/api/core/demo_mode_runner.py:414]:
```suggestion
    keep_api_key = os.environ.get("KEEP_READ_ONLY_BYPASS_KEY")
```

**Comment by Matvey-Kuk** [comment]
@coderabbitai resolve

**Pull request closed without merging**